cmd/bootstrap/cmd/constraints.go

```diff
@@ -35,14 +38,23 @@ func checkConstraints(partnerNodes, internalNodes []model.NodeInfo) {
 
 	ensureUniformNodeWeightsPerRole(all)
 
-	// check collection committee Byzantine threshold for each cluster
-	// for checking Byzantine constraints, the seed doesn't matter
-	_, clusters := constructClusterAssignment(partnerNodes, internalNodes, 0)
-	partnerCOLCount := uint(0)
-	internalCOLCount := uint(0)
-	for _, cluster := range clusters {
-		clusterPartnerCount := uint(0)
-		clusterInternalCount := uint(0)
+	// check collection committee threshold of internal nodes in each cluster
+	// although the assignmment is non-deterministic, the number of internal/partner
+	// nodes in each cluster is deterministic. The following check is only a sanity
+	// check about the number of internal/partner nodes in each cluster. The identites
+	// in each cluster do not matter for this sanity check.
+	_, clusters, err := constructClusterAssignment(partnerNodes, internalNodes)
+	if err != nil {
+		log.Fatal().Msgf("can't bootstrap because the cluster assignment failed: %s", err)
+	}
+	checkClusterConstraint(clusters, partners, internals)
+}
+
+// Sanity check about the number of internal/partner nodes in each cluster. The identites
+// in each cluster do not matter for this sanity check.
+func checkClusterConstraint(clusters flow.ClusterList, partners flow.IdentityList, internals flow.IdentityList) {
+	for i, cluster := range clusters {
```

Comment:
`constructClusterAssignment` will enforce this constraint on its own now, so we can remove this extra check.
- remove function `checkClusterConstraint`
- remove lines 41-50

Reply:
I have moved this check as a unit test, to make sure the clustering logic doesn't get updated by mistake.